### ISSUE TITLE
Upload and Merge not working on basic example

### ISSUE THREAD

**Issue by matthieus** [protegeproject/webprotege#496]
I am using WebProtégé 3.0.0 on a local installation.

Original ontology:
```
<?xml version="1.0"?>
<rdf:RDF xmlns="http://webprotege.stanford.edu/project/l1Z8CDZS3UplsBn3zeJw1#"
     xml:base="http://webprotege.stanford.edu/project/l1Z8CDZS3UplsBn3zeJw1"
     xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
     xmlns:owl="http://www.w3.org/2002/07/owl#"
     xmlns:xml="http://www.w3.org/XML/1998/namespace"
     xmlns:xsd="http://www.w3.org/2001/XMLSchema#"
     xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#">
    <owl:Ontology rdf:about="http://webprotege.stanford.edu/project/l1Z8CDZS3UplsBn3zeJw1"/>
    <owl:Class rdf:about="http://webprotege.stanford.edu/R9Jn34aUVD5Lm18IjLwpSZw">
        <rdfs:subClassOf rdf:resource="http://www.w3.org/2002/07/owl#Thing"/>
        <rdfs:label>bluh</rdfs:label>
    </owl:Class>
</rdf:RDF>
```
modified ontology created from downloading the existing one and modifying the one class label from "bluh" => "blah"):
```
<?xml version="1.0"?>
<rdf:RDF xmlns="http://webprotege.stanford.edu/project/l1Z8CDZS3UplsBn3zeJw1#"
     xml:base="http://webprotege.stanford.edu/project/l1Z8CDZS3UplsBn3zeJw1"
     xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
     xmlns:owl="http://www.w3.org/2002/07/owl#"
     xmlns:xml="http://www.w3.org/XML/1998/namespace"
     xmlns:xsd="http://www.w3.org/2001/XMLSchema#"
     xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#">
    <owl:Ontology rdf:about="http://webprotege.stanford.edu/project/l1Z8CDZS3UplsBn3zeJw1"/>
    <owl:Class rdf:about="http://webprotege.stanford.edu/R9Jn34aUVD5Lm18IjLwpSZw">
        <rdfs:subClassOf rdf:resource="http://www.w3.org/2002/07/owl#Thing"/>
        <rdfs:label>blah</rdfs:label>
    </owl:Class>
</rdf:RDF>
```

After uploading the above file through Project -> Upload and Merge, I get:

<img width="296" alt="merge summary" src="https://user-images.githubusercontent.com/329754/42208169-6f94924e-7ea3-11e8-9314-7bceca2b63ae.png">

And after pressing OK:

<img width="498" alt="error message after pressing ok" src="https://user-images.githubusercontent.com/329754/42208168-6f7d023c-7ea3-11e8-8692-98a54d938561.png">

Am I using this feature how it should be? Every test I ran ended with the same "An error has occurred. Please try again. null".

Looking at the logs I also saw the following stacktrace:
```
03-Jul-2018 01:27:36.060 SEVERE [http-nio-8080-exec-4] edu.stanford.bmir.protege.web.server.dispatch.impl.DispatchServiceExecutorImpl.execute An error occurred whilst executing an action
 java.lang.NullPointerException
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:770)
        at edu.stanford.bmir.protege.web.server.change.ChangeApplicationResult.<init>(ChangeApplicationResult.java:37)
        at edu.stanford.bmir.protege.web.server.project.ChangeManager.applyChanges(ChangeManager.java:295)
        at edu.stanford.bmir.protege.web.server.merge.MergeUploadedProjectActionHandler.applyChanges(MergeUploadedProjectActionHandler.java:95)
        at edu.stanford.bmir.protege.web.server.merge.MergeUploadedProjectActionHandler.execute(MergeUploadedProjectActionHandler.java:87)
        at edu.stanford.bmir.protege.web.server.merge.MergeUploadedProjectActionHandler.execute(MergeUploadedProjectActionHandler.java:41)
        at edu.stanford.bmir.protege.web.server.dispatch.impl.DispatchServiceExecutorImpl.execute(DispatchServiceExecutorImpl.java:71)
        at edu.stanford.bmir.protege.web.server.dispatch.impl.DispatchServiceImpl.executeAction(DispatchServiceImpl.java:50)
        at sun.reflect.GeneratedMethodAccessor35.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:587)
        at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:333)
        at com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:303)
        at edu.stanford.bmir.protege.web.server.app.WebProtegeRemoteServiceServlet.processCall(WebProtegeRemoteServiceServlet.java:61)
        at com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:373)
        at com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:660)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at edu.stanford.bmir.protege.web.server.filter.WebProtegeWebAppFilter.doFilter(WebProtegeWebAppFilter.java:75)
        at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
        at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
        at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199)
        at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
        at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:494)
        at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:137)
        at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
        at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:651)
        at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87)
        at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
        at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:407)
        at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
        at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:754)
        at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1376)
        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by matthieus** [comment]
Looks like this is happening only on the 3.0.0 released version. This issue is not happening when using the latest snapshot built from source.

Is there an ETA for a new official release?

**Comment by matthewhorridge** [comment]
@matthieus you're right – this has been fixed since 3.0.0.  We don't have an ETA for the new release, but I would have thought it would be in the next month or so.

**Comment by matthewhorridge** [comment]
Closing this because we have released 4.0.0-beta-1, where this is fixed
